### PR TITLE
Create authModal talon

### PR DESCRIPTION
**Pull request by sirugh**

### CODE REVIEW

File: packages/peregrine/lib/mixins/AuthModal/useAuthModal.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    const {
+        closeDrawer,
+        showCreateAccount,
+        showForgotPassword,
+        showMainMenu,
+        showMyAccount,
+        view
+    } = props;
+
+    const [username, setUsername] = useState('');
```
Comment (sirugh) [inline packages/peregrine/lib/mixins/AuthModal/useAuthModal.js]: This state isn't used in the hook anywhere, should it exist in the `AuthModal` component only?

Comment (sirugh) [reply packages/peregrine/lib/mixins/AuthModal/useAuthModal.js]: > but it's totally fine. I want the hook to do it

- Sanford, 2019

File: packages/venia-ui/lib/components/AuthModal/authModal.js [hunk 2]
```diff
@@ -1,51 +1,27 @@
...
-        async values => {
-            await createAccount(values);
-            showMyAccount();
-        },
-        [createAccount, showMyAccount]
-    );
+    const {
+        handleClose,
+        handleCreateAccount,
+        handleSignOut,
```
Comment (supernova-at) [inline packages/venia-ui/lib/components/AuthModal/authModal.js]: It strikes me suddenly that these may also be areas where developers may want to customize behaviors. Ex: send an analytic event, pop up a dialog, etc. in addition to the default behavior.

Totally out of scope for this PR but I'd be interested in any thoughts y'all have had on this front.

File: packages/peregrine/lib/talons/AuthModal/useAuthModal.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        showMyAccount,
+        view
+    } = props;
+
+    const [username, setUsername] = useState('');
+    const [{ currentUser }, { createAccount, signOut }] = useUserContext();
+
+    // if the user is authed, the only valid view is "MY_ACCOUNT"
+    useEffect(() => {
+        if (currentUser && currentUser.id && UNAUTHED_ONLY.includes(view)) {
```
Comment (supernova-at) [inline packages/peregrine/lib/talons/AuthModal/useAuthModal.js]: We probably don't even need that last conditional, right?
```suggestion
        if (currentUser && currentUser.id) {
```

File: packages/venia-ui/lib/components/AuthModal/authModal.js (hunk 2 above)
Comment (supernova-at) [reply packages/venia-ui/lib/components/AuthModal/authModal.js]: I love the consistency of the API coming back from the hook 🥇

File: packages/venia-ui/lib/components/AuthModal/authModal.js [hunk 4]
```diff
@@ -1,51 +1,27 @@
...
 
 import { mergeClasses } from '../../classify';
 import CreateAccount from '../CreateAccount';
 import ForgotPassword from '../ForgotPassword';
 import MyAccount from '../MyAccount';
 import SignIn from '../SignIn';
 import defaultClasses from './authModal.css';
-
-const UNAUTHED_ONLY = ['CREATE_ACCOUNT', 'FORGOT_PASSWORD', 'SIGN_IN'];
+import { useAuthModal } from '@magento/peregrine/lib/mixins/AuthModal/useAuthModal';
```
Comment (supernova-at) [inline packages/venia-ui/lib/components/AuthModal/authModal.js]: `mixins` has been renamed to `talons` 💪 🐦 💪 
```suggestion
import { useAuthModal } from '@magento/peregrine/lib/talons/AuthModal/useAuthModal';
```

File: packages/peregrine/lib/talons/AuthModal/useAuthModal.js (hunk 3 above)
Comment (sirugh) [reply packages/peregrine/lib/talons/AuthModal/useAuthModal.js]: Sure. I can change this.

**Comment by dpatil-magento** [comment]
QA Pass, need reviewer approval for latest commits.

File: packages/peregrine/lib/talons/AuthModal/useAuthModal.js (hunk 3 above)
Comment (supernova-at) [reply packages/peregrine/lib/talons/AuthModal/useAuthModal.js]: Did we end up needing it after all?

**Pull request merged**